lib/easyredmine_budget_quotas/time_entry_validation.rb

```diff
@@ -71,41 +74,45 @@ def check_if_budget_quota_valid
 
             assignable_hours = assignable_value/value_per_hour
 
-            # get current indx from comment
-            comment_id = self.comments.match(/(?<=\[)[0-9]{1,}/)
+            # get current index from comment
+            comment_id = self.comments.match(/(?<=\[)[0-9]{1,}/).to_id rescue nil
             if comment_id.nil?
               self.comments = "[1] #{self.comments}"
             else
               self.comments = "[#{comment_id.to_i+1}] #{self.comments.gsub(/\[[0-9]{1,}\]\ /, '')}"
             end
 
-
             # store values for next time entry and close current BudgetQuota
             @remaining_values_for_assignment = self.attributes.merge('hours' => self.hours - assignable_hours)
             current_bqs.first.update_column(:budget_quota_exceeded, true)
 
             # Assign currently applicable value
             self.hours = assignable_hours
-            assign_custom_field_value_for_ebq_budget_quota!(id: current_bqs.first.id, value: assignable_hours*value_per_hour*-1)
-          else
-            assign_custom_field_value_for_ebq_budget_quota!(id: current_bqs.first.id, value: will_be_spent*-1)
+
+            #assign_custom_field_value_for_ebq_budget_quota!(id: current_bqs.first.id, value: (assignable_hours*value_per_hour*-1).round(2))
+            # recal will_be_spent on new hours
+            will_be_spent = EasyMoneyTimeEntryExpense.compute_expense(self, project.calculation_rate_id)
```

Comment:
@florianeck 
fix rounding errors, so that the amount fits to the hours.round(2) * rate